apps/cowswap-frontend/vite.config.ts

```diff
@@ -0,0 +1,85 @@
+/// <reference types="vitest" />
+import { lingui } from '@lingui/vite-plugin'
+import react from '@vitejs/plugin-react-swc'
+import { defineConfig, loadEnv, searchForWorkspaceRoot } from 'vite'
+import macrosPlugin from 'vite-plugin-babel-macros'
+import { VitePWA } from 'vite-plugin-pwa'
+import svgr from 'vite-plugin-svgr'
+import viteTsConfigPaths from 'vite-tsconfig-paths'
+
+export default defineConfig(({ mode }) => {
+  const env = loadEnv(mode, process.cwd(), ['REACT_APP_'])
+
+  // expose .env as process.env instead of import.meta since jest does not import meta yet
+  const envWithProcessPrefix = Object.entries(env).reduce((prev, [key, val]) => {
+    return {
+      ...prev,
+      ['process.env.' + key]: JSON.stringify(val),
+    }
+  }, {})
+
+  return {
+    define: {
+      ...envWithProcessPrefix,
+    },
+
+    assetsInclude: ['**/*.md'],
+
+    cacheDir: '../../node_modules/.vite/cowswap-frontend',
+
+    server: {
+      port: 3000,
+      host: 'localhost',
+      fs: {
+        allow: [
+          // search up for workspace root
+          searchForWorkspaceRoot(process.cwd()),
+          // your custom rules
+          'apps/cowswap-frontend/src',
+          'libs',
+        ],
+      },
+    },
+
+    preview: {
+      port: 3000,
+      host: 'localhost',
+    },
+
+    resolve: {
+      alias: {
+        'node-fetch': 'isomorphic-fetch',
+      },
+    },
+
+    plugins: [
+      react({
+        plugins: [['@lingui/swc-plugin', {}]],
+      }),
+      viteTsConfigPaths({
+        root: '../../',
+      }),
+      macrosPlugin(),
+      lingui({
+        cwd: 'apps/cowswap-frontend',
+      }),
+      svgr(),
+      VitePWA({
+        registerType: 'autoUpdate',
+        injectRegister: 'auto',
+        devOptions: {
+          enabled: true,
+        },
+      }),
+    ],
+
+    // Uncomment this if you are using workers.
```

Comment:
Aren't we?

Reply:
Workers mean `Web workers` here.
For `Service workers` we use `import { VitePWA } from 'vite-plugin-pwa'`